clients/integrations/nemo-gym/tests/unit/test_provider.py

```
@@ -0,0 +1,479 @@
+    assert scripts[1].startswith("rm -f " + STAGING_PREFIX)
+    assert target.read_bytes() == b"tarball-bytes"
+
+
+async def test_download_file_missing_source_raises():
+    sandbox = FakeAsyncSandbox()
+    sandbox._commands.results.append(exec_result(stderr="cp: not found", exit_code=1))
+    provider = make_provider()
+    handle = make_handle(sandbox)
+
+    with pytest.raises(RuntimeError, match="not found"):
+        await provider.download_file(handle, "/data/missing", "/tmp/out")
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/out"

(S108)